adagrams/game.py

```diff
@@ -1,11 +1,168 @@
+import random
+
+# Dictionary of the letters with their quantities
+LETTER_POOL = {
+    "A": 9,
+    "B": 2,
+    "C": 2,
+    "D": 4,
+    "E": 12,
+    "F": 2,
+    "G": 3,
+    "H": 2,
+    "I": 9,
+    "J": 1,
+    "K": 1,
+    "L": 4,
+    "M": 2,
+    "N": 6,
+    "O": 8,
+    "P": 2,
+    "Q": 1,
+    "R": 6,
+    "S": 4,
+    "T": 6,
+    "U": 4,
+    "V": 2,
+    "W": 2,
+    "X": 1,
+    "Y": 2,
+    "Z": 1,
+}
+
+# Dictionary assigning point values to the letters
+LETTERS_VALUE = {
+    "A": 1,
+    "B": 3,
+    "C": 3,
+    "D": 2,
+    "E": 1,
+    "F": 4,
+    "G": 2,
+    "H": 4,
+    "I": 1,
+    "J": 8,
+    "K": 5,
+    "L": 1,
+    "M": 3,
+    "N": 1,
+    "O": 1,
+    "P": 3,
+    "Q": 10,
+    "R": 1,
+    "S": 1,
+    "T": 1,
+    "U": 1,
+    "V": 4,
+    "W": 4,
+    "X": 8,
+    "Y": 4,
+    "Z": 10,
+}
+
+
 def draw_letters():
-    pass
+
+    # Draw 10 letters randomly, considering the quantity limits
+    letter_list = list(LETTER_POOL.keys())
+    tiles = {}  # Dictionary with letter and count to avoid nested loops
+    letter_bank = [] 
+
+    while len(letter_bank) < 10:
+        random_number = random.randint(0, len(letter_list) - 1)
+        random_letter = letter_list[random_number]
+
+        if random_letter in tiles:
+            if tiles[random_letter] >= LETTER_POOL[random_letter]:
+                continue
+            tiles[random_letter] += 1
+        else:
+            tiles[random_letter] = 1
+            
+        letter_bank.append(random_letter)
+
+    return letter_bank
```

Comment:
Overall, this function looks good! It's a different approach than most that I've seen so far and it's quite clever! The one thing to point out is that while you are taking into account quantity limits, your code currently treats every letter as if they have the same chance of being picked! In reality though, there are more "A"s than "Q"s so the probability of picking an A would be much higher and your current algorithm doesn't take that into account! I will say this wasn't required and it wasn't a test case, so you are not required to include that, but if you have some extra time and would like to try an implementation that weights the letters, feel free to do so!